epicure-admin/src/app/components/generic-table/generic-table.component.html

```diff
@@ -0,0 +1,57 @@
+
+<div class="table-container">
+    <h2 class="table-title">{{ pageTitle }}s Table</h2>
+  <table mat-table [dataSource]="data" class="mat-elevation-z8">
+    
+    <!-- ID Column -->
+    <ng-container matColumnDef="_id">
+      <th mat-header-cell *matHeaderCellDef> ID </th>
+      <td mat-cell *matCellDef="let element"> {{element._id}} </td>
+    </ng-container>
+
+  <!-- Title Column -->
+  <ng-container matColumnDef="title">
+    <th mat-header-cell *matHeaderCellDef> Title </th>
+    <td mat-cell *matCellDef="let element"> {{element.title}} </td>
+  </ng-container>
+
+    <!-- Image Column -->
+    <ng-container matColumnDef="image">
+      <th mat-header-cell *matHeaderCellDef> Image </th>
+      <td mat-cell *matCellDef="let element">
+        <img [src]="element.image" alt="{{element.title}}" width="50" height="50">
+      </td>
+    </ng-container>
+
+    <!-- Dynamic columns -->
+    <ng-container *ngFor="let column of displayedColumns">
+      <ng-container *ngIf="column !== '_id' && column !== 'title' && column !== 'image' && column !== 'actions'">
```

Comment:
ngif here is negative with a lot of conditions, maybe use positive conditions?

Reply:
the positive condition will be longer then the negative, bc the component is generic, which means it contains inside her a lot of fields.